subsys/net/lib/lwm2m/lwm2m_obj_gateway.c

```diff
@@ -215,4 +215,4 @@ static int lwm2m_gw_init(void)
 	return ret;
 }
 
-SYS_INIT(lwm2m_gw_init, APPLICATION, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
+LWM2M_CORE_INIT(lwm2m_gw_init);
```

Comment:
Shouldn't this be a `LWM2M_OBJ_INIT`? I don't see Gateway object defined in the core spec.

Reply:
I was misguided by `	lwm2m_gw.is_core = true;`

Maybe I need to fix that as well.

Reply:
hmm, yes that seems wrong